### DISCUSSION TITLE
Providing an InstantSearchSTATICProvider for SSG

### DISCUSSION THREAD

**Discussion by hadminh** [algolia/instantsearch#5282] [Ideas]
Thanks for the new React Hooks library. I have been testing it and I must say that I am impressed by its performance. 

Are you planning to provide a Static Site Generation support? I have noticed that there is an `InstantSearchSSRProvider` and I was wondering if you are planning for a SSG provider like `InstantSearchSTATICProvider`?

**Comment by Haroenv** [comment]
the InstantSearchSSRProvider works for SSG too, I've tried it before. Where are you getting stuck? Do you have a sandbox with a replication of where you are currently?

**Reply by hadminh** [reply]
Thanks for the reply. Yes It does work by doing this but we don't have access to `req` inside getStaticProps(). If the url changes dynamically, I am not sure how to pass it. Sorry I currently don't have a sandbox.
```
export async function getStaticProps(context) {
  console.log(context);
  const url = 'http://localhost:3000/';
  const serverState = await getServerState(<HomePage url={url} />);
  return {
    props: {
      serverState,
      url
    }
  }
}
```

**Reply by Haroenv** [reply]
According to the API reference, the context contains the Next page parameters: https://nextjs.org/docs/api-reference/data-fetching/get-static-props#context-parameter. When you generate statically you don't have access to things like query parameters, as they aren't static
